pkg/services/internal/whatsonchain/dto/bsv_exchange_rate.go

```diff
@@ -0,0 +1,17 @@
+package dto
+
+// BSVExchangeRateResponse represents the exchange rate information for BSV
+// as returned by the WhatsOnChain API.
+//
+// It includes the timestamp of the rate, the exchange rate value itself,
+// and the fiat currency that the rate is quoted in.
+type BSVExchangeRateResponse struct {
```

Comment:
this way, you make it public in entire codebase (at least not published to the whole world)

I would prefer either of this (the order is not accidental):
1. keep those dto's private in a whatsonchain package
2. move dto to `internal/dto` which makes them internal to whatsonchain.